tiledb/sm/query/reader.cc

```diff
@@ -1540,15 +1547,37 @@ void Reader::fill_dense_coords_col_slab(
 
 Status Reader::filter_tiles(
     const std::string& attribute,
-    const std::vector<ResultTile*>& result_tiles) const {
+    const std::vector<ResultTile*>& result_tiles,
+    const std::vector<ResultCellSlab>& result_cell_slabs) const {
   STATS_FUNC_IN(reader_filter_tiles);
 
   auto var_size = array_schema_->var_size(attribute);
   auto num_tiles = static_cast<uint64_t>(result_tiles.size());
   auto encryption_key = array_->encryption_key();
 
+  // Build an association from the result tile to the cell slab ranges
+  // that it contains.
+  std::unordered_map<
+      ResultTile*,
+      std::forward_list<std::pair<uint64_t, uint64_t>>>
+      cs_ranges;
+  std::vector<ResultCellSlab>::const_reverse_iterator rit =
```

Comment:
to ensure that `forward_list` is ordered in ascending order, i'm reverse-iterating the slabs vector because `forward_list` only supports `emplace_front` (and not `emplace_back`). It's ugly but theoretically faster than using a doublely linked list or a vector.